SDWebImage.podspec

```diff
@@ -28,8 +28,9 @@ Pod::Spec.new do |s|
   s.default_subspec = 'Core'
 
   s.subspec 'Core' do |core|
-    core.source_files = 'SDWebImage/*.{h,m}', 'WebImage/SDWebImage.h'
+    core.source_files = 'SDWebImage/*.{h,m}', 'WebImage/SDWebImage.h', 'SDWebImage/Private/*.{h,m}'
     core.exclude_files = 'SDWebImage/MapKit/*.{h,m}'
+    core.private_header_files = 'SDWebImage/Private/*.h'
```

Comment:
I remember @bpoplauschi said private header may has issues?

Reply:
Actually, I tested and it's not. I and @kinarob achieve the agreement to follow other open-source frameworks' usage about private headers. The framework we choosed it's FaceBook's https://github.com/Instagram/IGListKit.

We follow the rules:

1. For internal class, keeing the file names with the class naming and mark it `Private Headers` in Xcode
2. For public classes' internal method, which can be called inside framework but not public, should be naming `class name + Internal.h`, and no `.m` files exist.
3. For public classes' internal category or property, should be naming `class name + Private.h` or `class name + category name.h`. This is a little different from the 2. because you have implementation codes here.

Reply:
The IGListKit private headers: https://github.com/Instagram/IGListKit/tree/master/Source/Internal. We follow the same structure as they are.

IGListKit is used widely in the World (even widely than SDWebImage itself) and I believe facebook team have a strong knowledge about the project management, code structure and the build tools (CocoaPods). This is the reason.

Reply:
Emm, I'm not oppose using `private_header_files`, I just mean @bpoplauschi said before, `private_header_files` may have issues(I don't know the reason), actually, I communicated one team of facebook before, the guys said in facebook they don't permit to use build tools like `CocoaPods`, they use [buck](https://buckbuild.com/), all open source from facebook have two version, OSS for open, and another for internal, something like below:

<img width="902" alt="image" src="https://user-images.githubusercontent.com/5061845/54473886-c4d1a780-4818-11e9-9dcb-1b787e9f89e8.png">

Reply:
If the company, use a copy of SDWebImage, but not using the official support build tools (We support 3: CocoaPods (official version with no modification), Carthage, Manual Install), they should take responsible for their build tools, but not force us to adopt their internal tools.

`private_header_files`, and which result as Xcode's offcial feature: 'Private Headers', it's the official way of all Apple platforms built feature for Objective-C/C/C++, so if their build tools don't support this syntax, it's their responsibility to fix.

Reply:
And. C-serious language, have a strong dependeny of the private headers && public headers. Although it's not a language-syntax because of C history, but all Apple's framework for iOS/tvOS/watchOS/macOS, benefit from the public/private/project headers for resource control.

Since our framework, written in Objective-C, but not Swift, we have no built-in language syntax for resource control, and should use this feature from Xcode.

Reply:
I might have said that, but I don't remember it :) If @dreampiggy says the private headers work, let's try them out. We can always revisit

Reply:
@bpoplauschi 😂 no matter, let's try it. ❤️ 

@dreampiggy Emm, seems you mess up Xcode and build tools, it's ok, let's try, I'm not opposed.